### PR TITLE
remove dead code related to DNA in `cm_restraints.cns` 

### PR DESCRIPTION
**Pull request by amjjbonvin**
Also added a protein-DNA cm-restrained docking test.
Checked all CNS output files for errors, none detected

You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md) and that you comply with the following criteria:

- [ ] You have sticked to Python. Please talk to us before adding other programming languages to HADDOCK3
- [X] Your PR is about CNS
- [ ] Your code is well documented: proper docstrings and explanatory comments for those tricky parts
- [ ] You structured the code into small functions as much as possible. You can use classes if there is a (state) purpose
- [X] Your code follows our coding style
- [X] You wrote CNS tests for the new code
- [X] `tox` or `pytest` tests pass. *Run `tox` command inside the repository folder*
- [X] `-test.cfg` examples execute without errors. *Inside `examples/` run `python run_tests.py -b`*
- [X] PR does not add any dependencies, unless permission granted by the HADDOCK team
- [X] PR does not break licensing
- [ ] Your PR is about writing documentation for already existing code :fire:
- [ ] Your PR is about writing tests for already existing code :godmode:

---

### CODE REVIEW

**Comment by rvhonorato** [comment]
Could you please change the title to reflect what was actually changed in the code? I'm usually changing after you, but for this one even I can't understand

**Comment by amjjbonvin** [comment]
Actually the issue description gives more details - the title cannot simply reflect the changes.

There were  undefined parameter for DNA (remaining of haddock2.X version) - but this was never noticed since we never did CM protein-DNA docking with haddock3.

The CNS part was simplified as the same behaviour is followed for ligand and nucleic acids.

File: examples/docking-protein-DNA/docking-protein-DNA-cmrest-test.cfg [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# ====================================================================
+# protein-DNA docking example
+
+# directory in which the scoring will be done
```
Comment (rvhonorato) [inline examples/docking-protein-DNA/docking-protein-DNA-cmrest-test.cfg:4]:
```suggestion
# TODO: Add this as an integration test
# directory in which the scoring will be done
```

**Review by rvhonorato — APPROVED** [review]
Keep in mind that while `examples/docking-protein-DNA/docking-protein-DNA-cmrest-test.cfg` executes a workflow that demonstrate the changes proposed in this PR, it is not actually testing for anything.

The proper way to test it would be to expand the integration tests for rigidbody and flexref, similar to what is being done here:

https://github.com/haddocking/haddock3/blob/b2dd2197d2ee996ebf1ef97a09e077b1ca026454/integration_tests/test_rigidbody.py#L134-L137

And add the assertions - I've suggested a `TODO` so this can be addressed later

File: examples/docking-protein-DNA/docking-protein-DNA-cmrest-test.cfg (hunk 1 above)
Comment (amjjbonvin) [reply examples/docking-protein-DNA/docking-protein-DNA-cmrest-test.cfg]: Ok - but it was needed to actually allow to check the CNS outputs for error messages. It is like all other tests in the examples dir.  Do we have already integration tests for all other test cfg files?

For integration tests we can have much simpler workflows that one check one specific module.

**Comment by amjjbonvin** [comment]
Ok - we will have then to define what to test on as for this and all other CNS-related tests, we need to check coordinates or energies in some way, allowing for some variation because of the chaotic nature of the computations. And there are so many different scenarios possible for each module.

**Comment by rvhonorato** [comment]
Yep - this is already being done, here for example checking for fnat (pending on #969)

https://github.com/haddocking/haddock3/blob/a644729d6161d576c6746f5168f222ea898a68c2/integration_tests/test_flexref.py#L69-L83

Other capri metrics can also be checked and other key values can also be validated within a range

**Review by rvhonorato — APPROVED** [review]

**Comment by rvhonorato** [comment]
oopsie, marked the wrong team - sorry guys 😅

**Review by VGPReys — APPROVED** [review]
Nice, this will help me writing the manual ;)

**Pull request merged**